File: lib/lang/es.js


function join_with_shared_prefix(a, b, joiner) {
  let i = 0;

  while(i !== a.length &&
        i !== b.length &&
        a.charCodeAt(i) === b.charCodeAt(i))
    ++i;

  while(i && a.charCodeAt(i - 1) !== 32)
    --i;

  return a + joiner + b.slice(i);
}

module.exports = {
  "clear": "despejado",
  "no-precipitation": "sin precipitaciones",
  "mixed-precipitation": "precipitación mixta",
  "possible-very-light-precipitation": "posibles lluvias ligeras",
  "very-light-precipitation": "lluvias ligeras",
  "possible-light-precipitation": "posibles lluvias ligeras",
  "light-precipitation": "precipitación ligera",
  "medium-precipitation": "precipitación",
  "heavy-precipitation": "fuerte precipitación",
  "possible-very-light-rain": "posible llovizna",
  "very-light-rain": "llovizna",
  "possible-light-rain": "posible lluvia ligera",
  "light-rain": "lluvia ligera",
  "medium-rain": "lluvia",
  "heavy-rain": "fuertes lluvias",
  "possible-very-light-sleet": "posible aguanieve ligera",
  "very-light-sleet": "aguanieve ligera",
  "possible-light-sleet": "posible aguanieve ligera",
  "light-sleet": "aguanieve ligera",
  "medium-sleet": "aguanieve",
  "heavy-sleet": "fuertes aguanieves",
  "possible-very-light-snow": "posible nevada ligera",
  "very-light-snow": "nevadas ligeras",
  "possible-light-snow": "posible nevada ligera",
  "light-snow": "nevadas ligeras",
  "medium-snow": "nevadas",
  "heavy-snow": "fuertes nevadas",
  "possible-thunderstorm": "tormentas posibles",
  "thunderstorm": "tormenta",
  "possible-medium-precipitation": "posibles precipitation",
  "possible-heavy-precipitation": "posibles fuertes precipitation",
  "possible-medium-rain": "posible lluvia",
  "possible-heavy-rain": "posible heavy lluvia",
  "possible-medium-sleet": "posible aguanieve",
  "possible-heavy-sleet": "posible fuertes aguanieve",
  "possible-medium-snow": "posible nevadas",
  "possible-heavy-snow": "posible fuertes nevadas",
  "possible-freezing-drizzle": "posible llovizna helada",
  "freezing-drizzle": "llovizna helada",
  "possible-light-freezing-rain": "posible nevadas lluvia helada",
  "light-freezing-rain": "nevadas lluvia helada",
  "possible-freezing-rain": "posible lluvia helada",
  "freezing-rain": "lluvia helada",
  "possible-heavy-freezing-rain": "posible fuertes lluvia helada",
  "heavy-freezing-rain": "fuertes lluvia helada",
  "possible-hail": "posible granizo",
  "hail": "granizo",
  "light-wind": "vientos suaves",
  "medium-wind": "ventoso",
  "heavy-wind": "peligrosamente ventoso",
  "low-humidity": "seco",
  "high-humidity": "húmedo",
  "fog": "niebla",
  "very-light-clouds": "mayormente despejado",
  "light-clouds": "parcialmente nublado",
  "medium-clouds": "mayormente nublado",
  "heavy-clouds": "nublado",
  "today-morning": "esta mañana",
  "later-today-morning": "después esta mañana",
  "today-afternoon": "esta tarde",
  "later-today-afternoon": "después esta tarde",
  "today-evening": "esta noche",
  "later-today-evening": "después esta noche",
  "today-night": "esta noche",
  "later-today-night": "después esta noche",
  "tomorrow-morning": "mañana por la mañana",
  "tomorrow-afternoon": "mañana por la tarde",
  "tomorrow-evening": "mañana por la noche",
  "tomorrow-night": "mañana por la noche",
  "morning": "por la mañana",
  "afternoon": "por la tarde",
  "evening": "por la noche",
  "night": "por la noche",
  "today": "hoy",
  "tomorrow": "mañana",
  "sunday": "el Domingo",
  "monday": "el Lunes",
  "tuesday": "el Martes",
  "wednesday": "el Miércoles",
  "thursday": "el Jueves",
  "friday": "el Viernes",
  "saturday": "el Sábado",
  "next-sunday": "el próximo Domingo",
  "next-monday": "el próximo Lunes",
  "next-tuesday": "el próximo Martes",
  "next-wednesday": "el próximo Miércoles",
  "next-thursday": "el próximo Jueves",
  "next-friday": "el próximo Viernes",
  "next-saturday": "el próximo Sábado",
  "minutes": "$1 min.",
  "fahrenheit": "$1\u00B0F",
  "celsius": "$1\u00B0C",
  "inches": "$1 in.",
  "centimeters": "$1 cm.",
  "less-than": "bajo $1",
  "and": function(a, b) {
    return join_with_shared_prefix(
      a,
      b,
      a.indexOf(",") !== -1 ? ", y " : " y ",
    );
  },
  "through": function(a, b) {
    return join_with_shared_prefix(a, b, " hasta ");
  },
  "with": "$1, con $2",
  "range": "$1\u2013$2",
  "parenthetical": "$1 ($2)",
  "for-hour": "$1 por la hora",
  "starting-in": "$1 comenzando en $2",
  "stopping-in": "$1 parando en $2",
  "starting-then-stopping-later": "$1 comenzando en $2, después parando en $3",
  "stopping-then-starting-later": "$1 parando en $2, comenzando de nuevo $3 después",
  "for-day": "$1 durante el día",
  "starting": "$1 comenzando $2",
  "until": "$1 hasta $2",
  "until-starting-again": "$1 hasta $2, comenzando otra vez $3",
  "starting-continuing-until": "$1 comenzando $2, continuando hasta $3",
  "during": "$1 $2",
  "for-week": "$1 durante la semana",
  "over-weekend": "$1 sobre el fin de semana",
  "temperatures-peaking": "temperaturas alcanzando un máximo de $1 $2",
  "temperatures-rising": "temperaturas llegando a $1 $2",
  "temperatures-valleying": "temperaturas alcanzando un mínimo de $1 $2",
  "temperatures-falling": "temperaturas cayendo a $1 $2",
  /* Capitalize the first letter of every word, except if that word is "y". */
  "title": function(str) {
    return str.replace(/\S+/g, function(word) {
      return word === "y" ?
        word :
        word.charAt(0).toUpperCase() + word.slice(1);
    });
  },
  /* Capitalize the first word of the sentence and end with a period. */
  "sentence": function(str) {
    /* Capitalize. */
    str = str.charAt(0).toUpperCase() + str.slice(1);

    /* Add a period if there isn't already one. */
    if(str.charAt(str.length - 1) !== ".")
      str += ".";

    return str;
  },
};

File: test_cases/es.py
# This would be one of your translations, as a Python dictionary
cases = {
    "Despejado": ["title", "clear"],
    "Posibles Lluvias Ligeras": ["title", "possible-very-light-precipitation"],
    "Precipitación Ligera": ["title", "very-light-precipitation"],
    "Posibles Lluvias Ligeras": ["title", "possible-light-precipitation"],
    "Precipitación Ligera": ["title", "light-precipitation"],
    "Precipitación": ["title", "medium-precipitation"],
    "Fuerte Precipitación": ["title", "heavy-precipitation"],
    "Posible Llovizna": ["title", "possible-very-light-rain"],
    "Llovizna": ["title", "very-light-rain"],
    "Posible Lluvia Ligera": ["title", "possible-light-rain"],
    "Lluvia Ligera": ["title", "light-rain"],
    "Lluvia": ["title", "medium-rain"],
    "Fuertes Lluvias": ["title", "heavy-rain"],
    "Posible Aguanieve Ligera": ["title", "possible-very-light-sleet"],
    "Aguanieve Ligera": ["title", "very-light-sleet"],
    "Posible Aguanieve Ligera": ["title", "possible-light-sleet"],
    "Aguanieve Ligera": ["title", "light-sleet"],
    "Aguanieve": ["title", "medium-sleet"],
    "Fuertes Aguanieves": ["title", "heavy-sleet"],
    "Posible Nevada Ligera": ["title", "possible-very-light-snow"],
    "Nevadas Ligeras": ["title", "very-light-snow"],
    "Posible Nevada Ligera": ["title", "possible-light-snow"],
    "Nevadas Ligeras": ["title", "light-snow"],
    "Nevadas": ["title", "medium-snow"],
    "Fuertes Nevadas": ["title", "heavy-snow"],
    "Ventoso": ["title", "medium-wind"],
    "Peligrosamente Ventoso": ["title", "heavy-wind"],
    "Niebla": ["title", "fog"],
    "Mayormente Despejado": ["title", "very-light-clouds"],
    "Parcialmente Nublado": ["title", "light-clouds"],
    "Mayormente Nublado": ["title", "medium-clouds"],
    "Nublado": ["title", "heavy-clouds"],
    "Seco y Vientos Suaves": ["title", ["and", "low-humidity", "light-wind"]],
    "Llovizna y Peligrosamente Ventoso": [
        "title",
        ["and", "very-light-rain", "heavy-wind"],
    ],
    "Húmedo y Parcialmente Nublado": [
        "title",
        ["and", "high-humidity", "light-clouds"],
    ],
    "Despejado por la hora.": ["sentence", ["for-hour", "clear"]],
    "Nevadas ligeras comenzando en 35 min.": [
        "sentence",
        ["starting-in", "very-light-snow", ["minutes", 35]],
    ],
    "Lluvia ligera parando en 15 min.": [
        "sentence",
        ["stopping-in", "light-rain", ["minutes", 15]],
    ],
    "Fuertes aguanieves comenzando en 20 min., después parando en 30 min.": [
        "sentence",
        [
            "starting-then-stopping-later",
            "heavy-sleet",
            ["minutes", 20],
            ["minutes", 30],
        ],
    ],
    "Lluvia parando en 25 min., comenzando de nuevo 8 min. después.": [
        "sentence",
        [
            "stopping-then-starting-later",
            "medium-rain",
            ["minutes", 25],
            ["minutes", 8],
        ],
    ],
    "Mayormente nublado durante el día.": ["sentence", ["for-day", "medium-clouds"]],
    "Aguanieve ligera comenzando por la mañana.": [
        "sentence",
        ["starting", "very-light-sleet", "morning"],
    ],
    "Ventoso hasta esta noche.": ["sentence", ["until", "medium-wind", "today-night"]],
    "Fuerte precipitación hasta por la tarde.": [
        "sentence",
        ["until", "heavy-precipitation", "afternoon"],
    ],
    "Vientos suaves por la tarde.": ["sentence", ["during", "light-wind", "afternoon"]],
    "Nevadas después esta noche y mañana por la mañana.": [
        "sentence",
        ["during", "medium-snow", ["and", "later-today-evening", "tomorrow-morning"]],
    ],
    "Fuertes lluvias hasta después esta mañana, comenzando otra vez esta noche.": [
        "sentence",
        ["until-starting-again", "heavy-rain", "later-today-morning", "today-evening"],
    ],
    "Nublado comenzando por la noche, continuando hasta por la noche.": [
        "sentence",
        ["starting-continuing-until", "heavy-clouds", "evening", "night"],
    ],
    "Aguanieve ligera después esta tarde y niebla mañana por la mañana.": [
        "sentence",
        [
            "and",
            ["during", "light-sleet", "later-today-afternoon"],
            ["during", "fog", "tomorrow-morning"],
        ],
    ],
    "Peligrosamente ventoso comenzando esta mañana, continuando hasta esta tarde, y aguanieve mañana por la mañana.": [
        "sentence",
        [
            "and",
            [
                "starting-continuing-until",
                "heavy-wind",
                "today-morning",
                "today-afternoon",
            ],
            ["during", "medium-sleet", "tomorrow-morning"],
        ],
    ],
    "Nublado comenzando después esta noche y fuertes nevadas mañana por la tarde.": [
        "sentence",
        [
            "and",
            ["starting", "heavy-clouds", "later-today-night"],
            ["during", "heavy-snow", "tomorrow-afternoon"],
        ],
    ],
    "Seco esta noche y precipitación ligera comenzando mañana por la noche, continuando hasta mañana por la noche.": [
        "sentence",
        [
            "and",
            ["during", "low-humidity", "today-night"],
            [
                "starting-continuing-until",
                "light-precipitation",
                "tomorrow-evening",
                "tomorrow-night",
            ],
        ],
    ],
    "Nevadas (5 in.) por la noche.": [
        "sentence",
        ["during", ["parenthetical", "medium-snow", ["inches", 5]], "night"],
    ],
    "Nevadas ligeras (2 cm.) después esta mañana.": [
        "sentence",
        [
            "during",
            ["parenthetical", "light-snow", ["centimeters", 2]],
            "later-today-morning",
        ],
    ],
    "Fuertes nevadas (8\u201312 in.) durante el día.": [
        "sentence",
        ["for-day", ["parenthetical", "heavy-snow", ["inches", ["range", 8, 12]]]],
    ],
    "Nevadas (bajo 1 cm.) por la tarde.": [
        "sentence",
        [
            "during",
            ["parenthetical", "medium-snow", ["less-than", ["centimeters", 1]]],
            "afternoon",
        ],
    ],
    "Sin precipitaciones durante la semana, con temperaturas alcanzando un máximo de 85\u00b0F mañana.": [
        "sentence",
        [
            "with",
            ["for-week", "no-precipitation"],
            ["temperatures-peaking", ["fahrenheit", 85], "tomorrow"],
        ],
    ],
    "Precipitación mixta sobre el fin de semana, con temperaturas llegando a 32\u00b0C el Jueves.": [
        "sentence",
        [
            "with",
            ["over-weekend", "mixed-precipitation"],
            ["temperatures-rising", ["celsius", 32], "thursday"],
        ],
    ],
    "Llovizna el Lunes, con temperaturas alcanzando un mínimo de 15\u00b0F el Viernes.": [
        "sentence",
        [
            "with",
            ["during", "very-light-rain", "monday"],
            ["temperatures-valleying", ["fahrenheit", 15], "friday"],
        ],
    ],
    "Nevadas ligeras el Martes y Miércoles, con temperaturas cayendo a 0\u00b0C el Domingo.": [
        "sentence",
        [
            "with",
            ["during", "light-snow", ["and", "tuesday", "wednesday"]],
            ["temperatures-falling", ["celsius", 0], "sunday"],
        ],
    ],
    "Precipitación hoy hasta el Sábado, con temperaturas alcanzando un máximo de 100\u00b0F el Lunes.": [
        "sentence",
        [
            "with",
            ["during", "medium-precipitation", ["through", "today", "saturday"]],
            ["temperatures-peaking", ["fahrenheit", 100], "monday"],
        ],
    ],
    "Tormentas posibles hasta el próximo Miércoles.": [
        "sentence",
        ["until", "possible-thunderstorm", "next-wednesday"],
    ],
}
